lock-api/src/main/java/com/palantir/lock/client/RemoteTimelockServiceAdapter.java

```diff
@@ -50,53 +52,73 @@ public final class RemoteTimelockServiceAdapter implements TimelockService, Auto
     private final TransactionStarter transactionStarter;
     private final CommitTimestampGetter commitTimestampGetter;
+    private final TimestampLeaseAcquirer timestampLeaseAcquirer;
+    private final MinLeasedTimestampGetter minLeasedTimestampGetter;
 
-    public RemoteTimelockServiceAdapter(
+    private RemoteTimelockServiceAdapter(
             NamespacedTimelockRpcClient rpcClient,
-            NamespacedConjureTimelockService conjureTimelockService,
+            NamespacedConjureTimelockService conjureClient,
             LockLeaseService lockLeaseService,
-            TransactionStarter transactionStarter,
-            CommitTimestampGetter commitTimestampGetter) {
+            RequestBatchersFactory batcherFactory,
+            TimestampLeaseAcquirer timestampLeaseAcquirer,
+            MinLeasedTimestampGetter minLeasedTimestampGetter) {
         this.rpcClient = rpcClient;
-        this.conjureTimelockService = conjureTimelockService;
+        this.conjureTimelockService = conjureClient;
         this.lockLeaseService = lockLeaseService;
-        this.transactionStarter = transactionStarter;
-        this.commitTimestampGetter = commitTimestampGetter;
+        this.transactionStarter = TransactionStarter.create(lockLeaseService, batcherFactory);
+        this.commitTimestampGetter = batcherFactory.createBatchingCommitTimestampGetter(lockLeaseService);
+        this.timestampLeaseAcquirer = timestampLeaseAcquirer;
+        this.minLeasedTimestampGetter = minLeasedTimestampGetter;
     }
 
-    private RemoteTimelockServiceAdapter(
+    public RemoteTimelockServiceAdapter(
             NamespacedTimelockRpcClient rpcClient,
             NamespacedConjureTimelockService conjureTimelockService,
-            LeaderTimeGetter leaderTimeGetter,
-            RequestBatchersFactory batcherFactory,
-            LockTokenUnlocker unlocker) {
+            LockLeaseService lockLeaseService,
+            TransactionStarter transactionStarter,
+            CommitTimestampGetter commitTimestampGetter,
+            TimestampLeaseAcquirer timestampLeaseAcquirer,
+            MinLeasedTimestampGetter minLeasedTimestampGetter) {
         this.rpcClient = rpcClient;
-        this.lockLeaseService = LockLeaseService.create(conjureTimelockService, leaderTimeGetter, unlocker);
-        this.transactionStarter = TransactionStarter.create(lockLeaseService, batcherFactory);
-        this.commitTimestampGetter = batcherFactory.createBatchingCommitTimestampGetter(lockLeaseService);
         this.conjureTimelockService = conjureTimelockService;
+        this.lockLeaseService = lockLeaseService;
+        this.transactionStarter = transactionStarter;
+        this.commitTimestampGetter = commitTimestampGetter;
+        this.timestampLeaseAcquirer = timestampLeaseAcquirer;
+        this.minLeasedTimestampGetter = minLeasedTimestampGetter;
     }
 
-    public static RemoteTimelockServiceAdapter create(
+    public static TimelockService create(
```

Comment:
Having both a static factory and a public constructor is pretty non-standard. When would we use the public constructor?

Reply:
Two bits:
- static factory and public constructor: this was the case before, I shifted arguments as necessary for my change and pushed the private one to the top. I am happy to address the existing cruft in this PR.
- breaks: totally missed this, addressing in a separate PR soon